src/lib.rs

```diff
@@ -116,13 +175,13 @@ pub mod pallet {
             .ok_or("Sub underflow")
             .unwrap();
 
+        log::debug!("disperse_reward:: reward total: {:?}", remainder);
         <RewardTotal<T>>::put(remainder as Value);
 
         for authority in auths {
-            let utxo = TransactionOutput {
-                value: share_value,
-                pub_key: *authority,
-            };
+            // TODO: where do we get the header info?
+            // TODO: are the rewards always of MLT token type?
```

Comment:
Does anyone have any idea how much effort it would be to change it so you can accept/validate different block reward tokens? Massive I guess. 
Worst case we enforce MLT for now and we'll have a DSA hardfork later so we can include token choices in that fork.

Reply:
It is quite difficult to predict how much effort it will take

Reply:
Yeah it was more a case of wondering if anyone had some across that chunk of code before

Reply:
I made the task in Trello about supporting rewards in the different tokens because it might appear a lot of places that we need to change. For example, after `fn validate_transaction` we are returning `ValidTransaction struct` this struct contain field `priority: TransactionPriority = u64` and this struct is inner Substrate type. It seems to be quite difficult to change this place but perhaps we can make our own struct for validation results. At the moment, we are using this field to collect information about rewards, but we should use for rewards something like this: `BTreeMap<crate::TokenID, Value>`. 

@BCarlaYap May you please tell us do we need In the future `ValidTransaction struct` in `fn spend`? And do you mind if I'll replace it with my own structure?

And if make supporting rewards in different tokens, as far as I understood at least we need to change storage design to save all those tokens values. 

I suggest returning here when the token system will allow us to make and send tokens.

Reply:
@sinitcin 
It's substrate who needs the `ValidTransaction` struct, so we can't  change it.

However, I think we don't necessarily need to house the reward inside `ValidTransaction`. I think it's okay to create our own storage struct for this.